packages/ui/src/views/repo/repo-commits/repo-commits-view.tsx

```diff
@@ -29,55 +30,97 @@ interface RepoCommitsViewProps {
   setSearchQuery: (query: string) => void
 }
 
-export const RepoCommitsView = (props: RepoCommitsViewProps) => {
-  const { t } = props.useTranslationStore()
+export const RepoCommitsView: FC<RepoCommitsViewProps> = ({
+  isFetchingCommits,
+  commitsList,
+  xNextPage,
+  xPrevPage,
+  page,
+  setPage,
+  selectBranchOrTag,
+  useTranslationStore,
+  useRepoBranchesStore,
+  searchQuery,
+  setSearchQuery
+}) => {
+  const { t } = useTranslationStore()
+
+  const FILTER_OPTIONS = getFilterOptions(t)
+  const SORT_OPTIONS = getSortOptions(t)
+  const SORT_DIRECTIONS = getSortDirections(t)
+  const filterHandlers = useFilters()
+
+  const isDirtyList = page !== 1
+
+  const handleResetFiltersAndPages = () => {
+    setPage(1)
+  }
```

Comment:
We probably want to reset to the 1st page on change of the filtering as well as clearing filters, otherwise the user could get stuck on a non-existent page if they change page with a vague query and then narrow down.